pandas/core/generic.py

```diff
@@ -4868,32 +4869,32 @@ def abs(self):
     @Appender(_shared_docs['describe'] % _shared_doc_kwargs)
     def describe(self, percentiles=None, include=None, exclude=None):
         if self.ndim >= 3:
-            msg = "describe is not implemented on on Panel or PanelND objects."
+            msg = "describe is not implemented on Panel or PanelND objects."
             raise NotImplementedError(msg)
+        elif self.ndim == 2 and self.columns.size == 0:
+            raise ValueError("Cannot describe a DataFrame without columns")
 
         if percentiles is not None:
             # get them all to be in [0, 1]
             self._check_percentile(percentiles)
+            # median should always be included
+            if 0.5 not in percentiles:
+                percentiles.append(0.5)
             percentiles = np.asarray(percentiles)
         else:
             percentiles = np.array([0.25, 0.5, 0.75])
 
-        # median should always be included
-        if (percentiles != 0.5).all():  # median isn't included
-            lh = percentiles[percentiles < .5]
-            uh = percentiles[percentiles > .5]
-            percentiles = np.hstack([lh, 0.5, uh])
+        # sort and check for duplicates
+        p_len = len(percentiles)
+        percentiles = np.unique(percentiles)
+        if len(percentiles) < p_len:
```

Comment:
Hmm, isn't it a bit of a waste to run `set` and don't use it later? I'd rather do:
```python
percentiles_unique = np.unique(percentiles)
if len(percentiles_unique) < len(percentiles):
    raise...
percentiles = percentiles_unique
```